core/src/main/java/site/ycsb/workloads/CoreWorkload.java

```diff
@@ -620,7 +657,22 @@ public boolean doInsert(DB db, Object threadstate) {
     Status status;
     int numOfRetries = 0;
     do {
-      status = db.insert(table, dbkey, values);
+      if (!isBatched) {
+        status = db.insert(table, dbkey, values);
+      } else {
+        if (batchKeysCount.get() < batchsize && opsDone.get() < recordcount) {
+          batchKeysList.add(dbkey);
+          batchValuesList.add(values);
+          batchKeysCount.incrementAndGet();
+          status = Status.BATCHED_OK;
+        } else {
+          status = db.batchInsert(table, batchKeysList, batchValuesList);
+          batchKeysList = new LinkedList<>();
```

Comment:
Why LinkedList? You know in advance the most common size of the list - it's a batchSize. Any why do you need to create new list instead of cleaning old one?

Reply:
updated